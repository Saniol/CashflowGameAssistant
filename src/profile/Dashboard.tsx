import React from 'react';
import {View, StyleSheet} from 'react-native';
import Lang from '../lang/pl';
import Backgrounds from '../global/Backgrounds';
import LayoutStyles from '../global/LayoutStyles';
import DisplayField from '../global/DisplayField';
import ActionRow from './ActionRow';
import ActionButton from './ActionButton';

const DashboardStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        flexWrap: 'wrap',
    },
    summaryColumn: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-around',
        paddingHorizontal: 16,
    },
});

const Dashboard: React.FC = () => (
    <View style={DashboardStyles.container}>
        <View style={LayoutStyles.rowsContainer}>
            <View style={[DashboardStyles.summaryColumn, Backgrounds.light]}>
                <DisplayField
                    label={Lang.dashboardProfession}
                    text="Nauczyciel"
                />
                <DisplayField label={Lang.dashboardAuditor} text="Martyna :)" />
                <DisplayField
                    label={Lang.dashboardDream}
                    text="Założyć własną galerię sztuki współczesnej
                w Krakowie"
                />
            </View>
            <View style={[DashboardStyles.summaryColumn, Backgrounds.light]}>
                <DisplayField label={Lang.dashboardCash} text="800$" />
                <DisplayField
                    label={Lang.dashboardMonthlyCashflow}
                    text="700$"
                />
                <DisplayField label={Lang.dashboardPassiveIncome} text="200$" />
            </View>
        </View>
        <View style={LayoutStyles.boldSeparator} />
        <ActionRow>
            <ActionButton color="yellow" title={Lang.dashboardCashflow} />
            <ActionButton color="green" title={Lang.dashboardNewTransaction} />
        </ActionRow>
        <ActionRow>
            <ActionButton color="red" title={Lang.dashboardNewExpense} />
            <ActionButton
                color="purple"
                title={Lang.dashboardOther}
                to="/other"
            />
        </ActionRow>
        <ActionRow>
            <ActionButton color="darkGreen" title={Lang.dashboardNewLoan} />
            <ActionButton
                color="blue"
                title={Lang.dashboardIncomeStatement}
                to="/income"
            />
        </ActionRow>
        <ActionRow>
            <ActionButton
                color="darkPurple"
                title={Lang.dashboardBalanceSheet}
                to="/balance"
            />
            <ActionButton
                color="orange"
                title={Lang.dashboardTransactions}
                to="/transactions"
            />
        </ActionRow>
    </View>
);

export default Dashboard;
