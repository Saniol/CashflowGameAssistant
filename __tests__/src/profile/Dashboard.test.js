import 'react-native';
import React from 'react';
import Dashboard from '../../../src/profile/Dashboard';
import renderWithContexts from '../../common/renderWithContexts';

describe('<Dashboard>', () => {
    renderWithContexts(Dashboard);
});
