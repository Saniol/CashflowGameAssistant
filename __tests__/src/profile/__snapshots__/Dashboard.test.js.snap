// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "flexWrap": "wrap",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-around",
            "paddingHorizontal": 16,
          },
          Object {
            "backgroundColor": "#F8E6B0",
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Zawód:
           
        </Text>
        <Text>
          Nauczyciel
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Audytor:
           
        </Text>
        <Text>
          Martyna :)
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Marzenie:
           
        </Text>
        <Text>
          Założyć własną galerię sztuki współczesnej w Krakowie
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-around",
            "paddingHorizontal": 16,
          },
          Object {
            "backgroundColor": "#F8E6B0",
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Gotówka:
           
        </Text>
        <Text>
          800$
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Miesięczny przepływ:
           
        </Text>
        <Text>
          700$
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Przepływ pasywny:
           
        </Text>
        <Text>
          200$
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#EEB35F",
        "height": 3,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFD024",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Wypłata
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#C5DF41",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Dokonaj transakcji
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF3822",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Nowy wydatek
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8B1263",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Inne
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#98B342",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Weź pożyczkę
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#828CCA",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Finanse
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#753558",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Pasywa i aktywa
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F65D05",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Historia
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
